### ISSUE TITLE
Select displayed timeline by tapping the header

### ISSUE THREAD

**Issue by Yowlen** [andstatus/andstatus#343]
Currently, we have to open the left sidebar to switch timelines. I think it'd be a lot easier if we could just tap the timeline title up at the top to bring up the timeline selection menu.

**Comment by yvolk** [comment]
We had a discussion on this recently: https://loadaverage.org/conversation/7574311

**Comment by Yowlen** [comment]
lol. Glad I'm not the only one. Makes it more likely to get implemented that way. ;)

**Comment by yvolk** [comment]
Which way? :-)

**Comment by Yowlen** [comment]
lol. Assuming it's a genuine question & not trolling, I just mean since there's more people who would like a feature, it's more likely to get implemented than if only one or two people requested it.

**Comment by yvolk** [comment]
I meant that in that discussion we talked about several ways to use one Title. Not for Timeline switch only. E.g.:
1. "Tapping on the title could directly open the selection list." - That will mean 2 clicks to switch a timeline instead of 3. Does that really that important? But if we would create a one click switch between Default timeline and any second timeline, you could simply make e.g. "Everything" your default timeline and if you need, switch to Mentions and back to Everything in one click. Switching to other timelines would still require 3 clicks. This looks more convenient for people, who uses not more than 2 timelines most of the time?!
2. On the "Quick timeline switch":
...we may set more than 2 timelines to switch on tapping a timeline title?! I mean that "Quck timeline switch" setting may be a multiselect list of timelines?! https://quitter.no/notice/982948
3. I prefer the "go to the top" option on the title (1 click) and direct access to timelines (mentions, dm, user) on the left panel (1 slide + 1 click)

**Comment by Yowlen** [comment]
Oh, sorry. I misunderstood the question. That's what I get for only skimming the post you linked. I appreciate your patience & explanation.

I didn't think of the other methods you listed, so I'm glad you mentioned them. I have to agree with you on number 3. That seems the best to me. It's possible to expand on it, too. People with small screens might have problems having to scroll through the list, so perhaps adding a "Favorite Timelines" list in the settings to allow the user to select which ones show in the sidebar will help out. The sidebar would just list the ones the user selects from that list along with a generic "More Timelines" option in the event the user wants to visit a timeline they don't normally visit.

**Comment by yvolk** [comment]
@Yowlen Thank you for the review.
 So you personally don't need that "Favorite timelines" feature?

**Comment by Yowlen** [comment]
Well, now that I have had time to clear my mind & come back to it, I just thought of a sort of compromise:
The favorites list the way I mentioned it could also be assigned to a new button to cycle through, while tapping the title itself will jump to the top like in your preferred method. In addition, tapping the title while it's already at the top could refresh the timeline, which would eliminate the need for the current refresh button.

So, basically, tap the title once to go to the top, then tap again to refresh the timeline. A button next to the title can be tapped to cycle through the favorite timeline list.

**Comment by yvolk** [comment]
AndStatus tip: ”Refresh from cache” may be needed in any place of a timeline, not at the top only, because sometimes images are downloaded later than messages. 
And if you have more than one account, newly downloaded messages of one account may appear below latest messages of another account.

Regarding another button - there is no place for it, especially on a small device. Before supporting Action Bar of Android 4, AndStatus had buttons across the whole top. One of them was exactly for timeline switch. Now placement of buttons is controlled by general Android library, this is why AndStatus looks like other apps...

**Comment by Yowlen** [comment]
Ah, ok. Well then, how about allowing the user to decide between "timeline switch" (option 2) & "go to top" (option 3)? Expanding the side panel to show all the timelines like you mention in option 3 essentially makes option 1 moot, anyway.

**Comment by yvolk** [comment]
Yes, it looks like we better adding a setting an option list ”Tap on a Timeline behaviour” with these values:
”Go to the Top” (default)
”Timeline selection”
”(disabled)"

**Comment by yvolk** [comment]
Added "Tap on a Timeline Title behaviour" option, which allows to select most convenient action between: "Switch to Default Timeline" (default), "Go to the Top" and "Select a Timeline". Actually, "Switch to Default Timeline" works the same way as  "Select a Timeline" if we're already on the Default Timeline.

To be released in v.26.0
![screenshot_2016-04-17-16-15-48](https://cloud.githubusercontent.com/assets/207716/14587210/e53e8904-04b7-11e6-9532-de05ca96f0a9.png)

**Comment by bulletbites** [comment]
Omg this already works like a charm! Good job!
